src/ui/UserProfile.h

```diff
@@ -109,15 +114,25 @@ class UserProfile : public QObject
         // Q_INVOKABLE void ignoreUser();
         Q_INVOKABLE void kickUser();
         Q_INVOKABLE void startChat();
+        Q_INVOKABLE void changeUsername(QString username);
+        Q_INVOKABLE void allowUsernameEditing(bool allow);
 
 signals:
         void userStatusChanged();
+        void usernameEditingChanged();
+        void displayNameChanged();
+        void globalUsernameRetrieved(const QString &globalUser);
+
+protected slots:
+        void setGlobalUsername(const QString &globalUser);
 
 private:
         QString roomid_, userid_;
+        QString globalUsername;
         DeviceInfoModel deviceList_;
-        bool isUserVerified = false;
-        bool hasMasterKey   = false;
+        bool isUserVerified  = false;
+        bool hasMasterKey    = false;
+        bool usernameEditing = false;
```

Comment:
It would be nice if you could add an underscore to the end of these variables (again, to match the coding style).

Reply:
Not really, we don't use that consistently enough, so it makes not much sense to nitpick here :D

Reply:
isUserVerified is used in the qml file, maybe that's why there was no underscore at the end of those variables?
Anyway what's the consensus here, should I or should I not add the underscore?

Reply:
It really does not matter, if there are no shadowed variables :D